### PR TITLE
feat:search and sort decks

### LINKED ISSUE #71: Order decks after creation date
When a deck is created, a time stamp should be saved. And when they are shown they should be sorted based on this.

https://stackoverflow.com/a/51847047

### PR DESCRIPTION
**Pull request by blendaf**
fixes: #71

### CODE REVIEW

File: components/browse-decks/BrowseDecks.js [hunk 1]
```diff
@@ -29,19 +29,59 @@ export default ({ decks, createDeck, addDeck }) => {
...
-        {Object.entries(decks).map(([id, deck]) => (
-          <Link key={id} href={`/view-deck?id=${id}`} passHref>
-            <DeckBox id={id} edit={uid === deck.uid}>
-              {deck.name}
-            </DeckBox>
-          </Link>
-        ))}
-      </DeckWrapper>
-    </DeckView>
+    <>
```
Comment (juliuscc) [inline components/browse-decks/BrowseDecks.js]: This fragment is not necessary.

File: components/browse-decks/DeckSearch.js [hunk 2]
```diff
@@ -21,45 +20,81 @@ const SearchBox = styled.input`
...
   padding: 10px 20px;
   border: 1px solid ${({ theme }) => theme.colors.border};
-  border-radius: 10px 0px 0px 10px;
+  border-radius: 10px;
   flex: 1;
   font-size: 18px;
+  :focus {
+    outline: none;
+    -webkit-box-shadow: 0px 0px 10px 2px ${({ theme }) => theme.colors.border};
+    -moz-box-shadow: 0px 0px 10px 2px ${({ theme }) => theme.colors.border};
```
Comment (juliuscc) [inline components/browse-decks/DeckSearch.js]: I am quite sure that styled-components autogenerates these flags. I don't think we should add them.

File: package.json [hunk 3]
```diff
@@ -23,7 +23,7 @@
     "awesome-only-resolves-last-promise": "^1.0.3",
     "firebase": "^7.9.1",
     "isomorphic-unfetch": "^3.0.0",
-    "next": "9.2.2",
+    "next": "^9.3.4",
```
Comment (juliuscc) [inline package.json]: Let's do this in another PR

File: components/browse-decks/DeckBox.js [hunk 4]
```diff
@@ -46,6 +46,15 @@ const DeckBox = styled.a`
   :disabled {
     cursor: not-allowed;
   }
+
+  :focus {
+    outline: none;
+    -webkit-box-shadow: 0px 0px 10px 2px
+      ${({ theme }) => theme.colors.primaryMuted};
+    -moz-box-shadow: 0px 0px 10px 2px
```
Comment (juliuscc) [inline components/browse-decks/DeckBox.js]: I don't think we should add flags. Most browsers have stopped relying on them, and we risk having different styles for different browsers if we forget to update one of these.

Comment (blendaf) [reply components/browse-decks/DeckBox.js]: fixed already

File: components/browse-decks/ViewBar.js [hunk 5]
```diff
@@ -17,13 +17,19 @@ const ViewBarBox = styled.div`
 const ViewLinks = styled.button`
   background-color: ${({ theme }) => theme.colors.focusBackground};
   border: none;
-  width: 100px;
-  height: 50px;
+  height: 80px;
```
Comment (juliuscc) [inline components/browse-decks/ViewBar.js:20]: Nice!

File: components/exchange/SearchBar.js [hunk 6]
```diff
@@ -79,15 +85,23 @@ const CharacterSelect = styled.div`
...
   cursor: pointer;
 
   @media screen and (max-width: ${screenSizes.tablet.max}) {
     padding: 0 5px;
   }
+  :focus {
+    outline: none;
+    -webkit-box-shadow: 0px 0px 10px 2px ${({ theme }) => theme.colors.border};
+    -moz-box-shadow: 0px 0px 10px 2px ${({ theme }) => theme.colors.border};
+    box-shadow: 0px 0px 10px 2px ${({ theme }) => theme.colors.border};
```
Comment (juliuscc) [inline components/exchange/SearchBar.js]: This shadow is used a lot. It might be a good idea to place it in the theme. Like
```js
...
 shadows: { outline: '0px 0px 10px 2px #SOMETHING' }
...
```

File: components/ui-fragments/Input.js [hunk 7]
```diff
@@ -13,17 +13,31 @@ export const Input = styled.input`
 const SearchInputWrapper = styled.div`
   width: 100%;
   display: flex;
+  
+  }
```
Comment (juliuscc) [inline components/ui-fragments/Input.js]: This looks like a mistake

File: components/ui-fragments/Input.js [hunk 8]
```diff
@@ -18,12 +18,20 @@ const SearchInputWrapper = styled.div`
 const HalfInput = styled(Input)`
   border-radius: 10px 0 0 10px;
   flex: 1;
+  :focus {
+    outline: none;
+    box-shadow: 0px 0px 10px 2px ${({ theme }) => theme.colors.border};
```
Comment (juliuscc) [inline components/ui-fragments/Input.js:23]: This shadow is used a lot. It might be a good idea to place it in the theme. Like
```js
...
 shadows: { outline: '0px 0px 10px 2px #SOMETHING' }
...
```

Comment (juliuscc) [reply components/ui-fragments/Input.js:23]: What do you think?

Comment (blendaf) [reply components/ui-fragments/Input.js:23]: will fix in another pr

**Review by juliuscc — APPROVED** [review]
